src/app/tests/integration/chip_im_initiator.cpp

```diff
@@ -293,8 +293,143 @@ void HandleWriteComplete()
 
     printf("Write Response: %" PRIu64 "/%" PRIu64 "(%.2f%%) time=%.3fms\n", gWriteRespCount, gWriteCount,
            static_cast<double>(gWriteRespCount) * 100 / gWriteCount, static_cast<double>(transitTime) / 1000);
+}
+
+void Shutdown()
+{
+    chip::app::InteractionModelEngine::GetInstance()->Shutdown();
+    ShutdownChip();
+}
+
+void CommandRequestTimerHandler(chip::System::Layer * systemLayer, void * appState, CHIP_ERROR error)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (gCommandRespCount != gCommandCount)
+    {
+        printf("No response received\n");
+
+        // Set gCommandRespCount to gCommandCount to start next iteration if there is any.
+        gCommandRespCount = gCommandCount;
+    }
+
+    if (gCommandRespCount < kMaxCommandMessageCount)
+    {
+        chip::app::CommandSender * commandSender;
+        err = chip::app::InteractionModelEngine::GetInstance()->NewCommandSender(&commandSender);
+        SuccessOrExit(err);
+
+        err = SendCommandRequest(commandSender);
+        SuccessOrExit(err);
+
+        err = chip::DeviceLayer::SystemLayer.StartTimer(gMessageIntervalSeconds * 1000, CommandRequestTimerHandler, NULL);
```

Comment:
So in this new setup each test will definitely run for `gMessageIntervalSeconds` instead of completing when the response is received, right?  Is that intended?

Reply:
Yes, it is intended. We always start new test after interval gMessageIntervalSeconds

Reply:
Alright.  Seems like it will make the test run much slower, but maybe that's ok....